src/core/load-fragment.c

```diff
@@ -6542,3 +6543,42 @@ int config_parse_log_filter_patterns(
 
         return 0;
 }
+
+int config_parse_open_file(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(open_file_freep) OpenFile *of = NULL;
+        OpenFile **head = ASSERT_PTR(data);
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                while ((of = *head)) {
+                        LIST_REMOVE(open_files, *head, of);
+                        open_file_free(of);
```

Comment:
Let's do `of = open_file_free(of)` here to be safe. Otherwise `of` might temporarily point to freed memory that will be freed again when the cleanup function is called.